subsys/net/lib/aws_iot/src/aws_iot.c

```diff
@@ -112,6 +112,11 @@ static aws_iot_evt_handler_t module_evt_handler;
 static atomic_t disconnect_requested;
 static atomic_t connection_poll_active;
 
+/* Flag that indicates if a the client is disconnected from the
+ * AWS IoT broker, or not.
+ */
+static atomic_t aws_iot_disconnected;
```

Comment:
This should be set to 1 on initialization, or changed to `aws_iot_connected` either here or on init. Now it indicates that the cloud is connected from the start.

Reply:
The variable is now initialized to 1.